php/WP_CLI/Runner.php

```diff
@@ -376,26 +376,12 @@ public function find_command_to_run( $args ) {
 			$subcommand = $command->find_subcommand( $args );
 
 			if ( ! $subcommand ) {
-				if ( count( $cmd_path ) > 1 ) {
-					$child       = array_pop( $cmd_path );
-					$parent_name = implode( ' ', $cmd_path );
-					$suggestion  = $this->get_subcommand_suggestion( $child, $command );
-
-					if ( 'network' === $parent_name && 'option' === $child ) {
-						$suggestion = 'meta';
-					}
-
-					return sprintf(
-						"'%s' is not a registered subcommand of '%s'. See 'wp help %s' for available subcommands.%s",
-						$child,
-						$parent_name,
-						$parent_name,
-						! empty( $suggestion ) ? PHP_EOL . "Did you mean '{$suggestion}'?" : ''
-					);
+				// Suggest appropriate commands for taxonomy or post type.
+				if ( $this->is_taxonomy( $cmd_path[0] ) ) {
+					WP_CLI::error( "Did you mean 'wp term <command> ?" );
+				} elseif ( $this->is_post_type( $cmd_path[0] ) ) {
+					WP_CLI::error( "Did you mean 'wp post <command> ?" );
```

Comment:
I think this should be wrapped in a `if ( $this->wp_exists() && $this->wp_is_readable() ) {}`

Otherwise we try to load WordPress all the time, which is not great, as then you get output like this for _every_ mistyped command:

```
$ wp foo
Warning: No WordPress installation found. If the command 'foo' is in a plugin or theme, pass --path=`path/to/wordpress`.
Error: 'foo' is not a registered wp command. See 'wp help' for available commands.
```

Reply:
So it could be like this:

```suggestion
				if ( $this->wp_exists() && $this->wp_is_readable() ) {
					$this->load_wordpress();

					// Suggest appropriate commands for taxonomy or post type.
					if ( $this->is_taxonomy( $cmd_path[0] ) ) {
						WP_CLI::error( "Did you mean 'wp term <command> ?" );
					} elseif ( $this->is_post_type( $cmd_path[0] ) ) {
						WP_CLI::error( "Did you mean 'wp post <command> --post-type=$cmd_path[0]?" );
					}
```